### PR TITLE
[ThemeBundle] Ability to configure theme locations

### PR DESCRIPTION
**Pull request by pamil**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes (default value changed from `%kernel.root.dir%` to `%kernel.root_dir%/themes`)
| Deprecations? | no
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Sylius/Bundle/ThemeBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        $rootNode
+            ->addDefaultsIfNotSet()
+            ->fixXmlConfig('location')
+                ->children()
+                    ->arrayNode('locations')
+                        ->requiresAtLeastOneElement()
+                        ->performNoDeepMerging()
+                        ->defaultValue(['%kernel.root_dir%/themes'])
+                        ->prototype('scalar')
```
Comment (tristanbes) [inline src/Sylius/Bundle/ThemeBundle/DependencyInjection/Configuration.php]: aren't some "end()" node missing ? indentation feels weird at the end :dango:

Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/DependencyInjection/Configuration.php]: `->end()` is not necessary as "closing open nodes", it only returns the parent node. Anyway, TDD for configurations and extensions seems great and it has to work correctly :)

File: src/Sylius/Bundle/ThemeBundle/Tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            ]
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_uses_the_last_theme_locations_passed_and_rejects_the_other_ones()
+    {
+        $this->assertProcessedConfigurationEquals(
```
Comment (bendavies) [inline src/Sylius/Bundle/ThemeBundle/Tests/DependencyInjection/ConfigurationTest.php]: very minor, but i find this formatting pretty grating.
How about:
```
$this->assertProcessedConfigurationEquals(
    [
        ['locations' => ['/custom/path', '/custom/path2']],
        ['locations' => ['/last/custom/path']],
    ],
    [
        'locations' => ['/last/custom/path'],
    ]
);
```

Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/Tests/DependencyInjection/ConfigurationTest.php]: :+1:, I worked on processing theme configuration and these arrays took enormously much space.

**Comment by pjedrzejewski** [comment]
@pamil Will you apply @bendavies's suggestion here or in #3956, I could merge this one, I guess. :)

**Comment by pamil** [comment]
@pjedrzejewski done :)

**Comment by pjedrzejewski** [comment]
:+1:

**Pull request merged**